finagle-http/src/main/scala/com/twitter/finagle/Http.scala

```diff
@@ -497,6 +497,35 @@ object Http extends Client[Request, Response] with HttpRichClient with Server[Re
     def withStreaming(fixedLengthStreamedAfter: StorageUnit): Server =
       configured(http.param.Streaming(fixedLengthStreamedAfter))
 
+    /**
+     * Enable/disable HA proxy protocol v1|v2
+     *
+     * Typically HTTP server is placed behind load balancer and therefore
+     * remote address of the request is pointing to it. In such case original
+     * client source address and destination port of the request is lost.
+     *
+     * Enable HA proxy protocol only if load balancer supports and pre-append requests
+     * with proxy protocol message v1|v2.
+     *
+     * @note Handling HA proxy protocol is currently NOT supported for:
+     *       - HTTP protocol version 2
+     *       - HTTP1.1 streaming
+     *
+     * @note Enabling HA Proxy protocol will disable HTTP2 and disable streaming for HTTP1.1
+     *
+     * @see [[https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt]]
+     */
+    def withHAProxyProtocol(enabled: Boolean): Server = {
+      if (enabled) {
+        this
+          .withNoHttp2
+          .withStreaming(false)
+          .configured(http.param.HAProxyProtocol(enabled))
+      } else {
+        configured(http.param.HAProxyProtocol(enabled))
+      }
```

Comment:
Hm, this is a bit weird since we take assumptions. Could we check if configuration has http2/streaming set and if it has we can throw exception that we can instantiate this protocol/server

Reply:
That is a good point. Let me check if and how it can be done.

Reply:
Looks nice